hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/linux/resources/CGroupsV2HandlerImpl.java

```diff
@@ -95,10 +95,21 @@ protected List<CGroupController> getCGroupControllers() {
   }
 
   @Override
-  protected Map<String, Set<String>> parsePreConfiguredMountPath() throws IOException {
+  protected Map<String, Set<String>> parsePreConfiguredMountPath() {
     Map<String, Set<String>> controllerMappings = new HashMap<>();
-    controllerMappings.put(this.cGroupsMountConfig.getV2MountPath(),
-        readControllersFile(this.cGroupsMountConfig.getV2MountPath()));
+    try {
+      controllerMappings.put(this.cGroupsMountConfig.getV2MountPath(),
+          readControllersFile(this.cGroupsMountConfig.getV2MountPath()));
+    } catch (IOException e) {
+      // Failing to read the cgroup.controllers file in the preconfigured might mean
+      // that the node is not using cgroup v2, or no cgroup v2 hierarchy is mounted
+      // under the specified path. If the node is using v1 we will fall back to cgroup v1
+      // in ResourceHandlerModule.initializeCGroupHandlers. If the cgroup v2 hierarchy is not mounted
+      // and no cgroup v1 hierarchy is mounted, we will fail to start the node manager.
```

Comment:
NIT: Maybe we can add a trace log with the exception if someone is interested in the stack trace?

Reply:
Added.